test/mx/test_transform_equals.py

```diff
@@ -0,0 +1,219 @@
+            time_feature.MonthOfYear(),
+        ],
+        pred_length=10,
+    )
+
+    tran5 = serde.flat.clone(tran4)
+    tran6 = serde.flat.clone(
+        tran4,
+        {
+            "time_features": [
+                time_feature.DayOfWeek(),
+                time_feature.DayOfMonth(),
+            ]
+        },
+    )
+
+    assert equals(tran4, tran5)
+    assert not equals(tran4, tran6)
+
+
+def test_filter_transformation():
+
+    prediction_length = 10
+    tran1 = transform.FilterTransformation(
+        lambda x: x["target"].shape[-1] > prediction_length
+    )
+    # serde.flat.clone(tran1) does not work on
+    tran2 = transform.FilterTransformation(
+        lambda x: x["target"].shape[-1] > prediction_length
+    )
+    tran3 = transform.FilterTransformation(
+        condition=lambda x: x["target"].shape[-1] < prediction_length
+    )
+
+    assert equals(tran1, tran2)
+    assert not equals(tran1, tran3)
+
+
+def test_sampler():
+    sampler = transform.ExpectedNumInstanceSampler(num_instances=4)
+    sampler2 = serde.flat.clone(sampler, {"num_instances": 5})
+    assert equals(sampler, serde.flat.clone(sampler))
+    assert not equals(sampler, sampler2)
+
+    sampler3 = transform.ContinuousTimeUniformSampler(num_instances=4)
+    assert equals(sampler3, serde.flat.clone(sampler3))
+    sampler4 = serde.flat.clone(sampler3, {"num_instances": 5})
+    assert not equals(sampler3, sampler4)
+
+
+def test_instance_splitter():
+
+    splitter = transform.InstanceSplitter(
+        target_field=FieldName.TARGET,
+        is_pad_field=FieldName.IS_PAD,
+        start_field=FieldName.START,
+        forecast_start_field=FieldName.FORECAST_START,
+        instance_sampler=transform.ExpectedNumInstanceSampler(num_instances=4),
+        past_length=100,
+        future_length=10,
+        time_series_fields=["dynamic_feat", "observed_values"],
+    )
+
+    assert equals(splitter, serde.flat.clone(splitter))
+    assert not equals(
+        splitter,
+        serde.flat.clone(
+            splitter,
+            {
+                "instance_sampler": transform.ExpectedNumInstanceSampler(
+                    num_instances=5
+                )
+            },
+        ),
+    )
+
+
+def test_continuous_time_splitter():
+    splitter = transform.ContinuousTimeInstanceSplitter(
+        past_interval_length=1,
+        future_interval_length=1,
+        instance_sampler=transform.ContinuousTimePointSampler(),
+    )
+
+    splitter2 = transform.ContinuousTimeInstanceSplitter(
+        past_interval_length=1,
+        future_interval_length=1,
+        instance_sampler=transform.ContinuousTimePointSampler(),
+    )
+
+    splitter3 = transform.ContinuousTimeInstanceSplitter(
+        past_interval_length=1,
+        future_interval_length=1,
+        instance_sampler=transform.ContinuousTimePointSampler(min_past=1.0),
+    )
+    assert equals(splitter, splitter2)
+    assert not equals(splitter, splitter3)
+
+
+def test_batch():
```

Comment:
Shouldn't this be tested as part of the loader?

Reply:
Moved.